Add support for DiT ONNX export

xenova opened the pull request:
# What does this PR do?

Not sure if this is actually needed, since DiT uses the same architecture as BeiT, but it adds it to the list 👍 
<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum/pr_1612). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

fxmarty commented:
@xenova 

Can you fix
```
FAILED exporters/onnx/test_onnx_export.py::OnnxExportTestCase::test_all_models_tested - AssertionError: Not testing all models. Missing models: {'dit'}
2023-12-24T18:41:32.3343756Z FAILED onnxruntime/test_modeling.py::ORTModelForImageClassificationIntegrationTest::test_pipeline_ort_model_01_dit - KeyError: 'dit'
2023-12-24T18:41:32.3344416Z FAILED onnxruntime/test_modeling.py::ORTModelForImageClassificationIntegrationTest::test_compare_to_transformers_01_dit - KeyError: 'dit'
```

xenova commented:
Converted to draft and will mark as ready when I've added back the unit tests. @fxmarty once again, it requires a tiny model - do you have a script to easily create these?
